### PR TITLE
Add support for torch/_tensor.py @_handle_torch_function_and_wrap_type_error_to_not_implemented marked methods

### PR DESCRIPTION
**Pull request by voznesenskym**
Tested with `compute-sanitizer --tool memcheck pytest test/test_torchinductor_opinfo.py -k rmatmul___cuda_float16`

### CODE REVIEW

**Comment by mlazos** [comment]
Why can't these functions be built as TorchVariables? `TorchDunderFunctions` are exactly this [else clause](https://github.com/pytorch/torchdynamo/blob/ddec1bd1ae511ff6fe11e0ecf4d24e1656bcaf79/torchdynamo/variables/torch.py#L223)

File: torchdynamo/variables/builder.py [hunk 1]
```diff
@@ -352,6 +353,11 @@ def index_source(key):
                 value, guards=make_guards(GuardBuilder.FUNCTION_MATCH)
             )
         elif istype(value, types.FunctionType):
+            if value in tensor_dunder_fns:
```
Comment (mlazos) [inline torchdynamo/variables/builder.py:356]: Can we add these to the `is_allowed` list, and have it handled by the existing clause?

Comment (mlazos) [reply torchdynamo/variables/builder.py:356]: Resolved offline, these are builtins which bridge to a torch op, I'm okay with special casing it here.

Comment (voznesenskym) [reply torchdynamo/variables/builder.py:356]: discussed offline, will keep as is.

File: torchdynamo/variables/functions.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
 from ..source import AttrSource
 from ..source import GetItemSource
 from ..utils import make_cell
+from ..utils import proxy_args_kwargs
```
Comment (mlazos) [inline torchdynamo/variables/functions.py:19]: Unused import?

**Review by mlazos — APPROVED** [review]

**Comment by voznesenskym** [comment]
> This job appears to have stopped responding, try re-running it on the circleci job. Boo.

**Pull request merged**